pkg/agent/flowexporter/connections/l7_listener.go

```diff
@@ -0,0 +1,223 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connections
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/netip"
+	"os"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/config"
+	"antrea.io/antrea/pkg/agent/flowexporter"
+
+	// "antrea.io/antrea/pkg/agent/util"
+	k8sutil "antrea.io/antrea/pkg/util/k8s"
+	"antrea.io/antrea/pkg/util/podstore"
+)
+
+type PodL7VisibilityAttrGetter interface {
+	IsL7VisibilityRequested(podNN string, ingress bool) bool
+}
+
+// L7ProtocolFields holds layer 7 protocols supported
+type L7ProtocolFields struct {
+	http []map[int32]*Http
```

Comment:
It's strange to have a slice of map, each has a single key.
Do we really need the TxID? Does it mean anything to users or help anything when processing the records?

Reply:
TxID is used for persistent HTTP connections, also txID would help us to deduplicate and append the new events in case of persistent HTTP

Reply:
then why isn't it just a map?
```
http map[int32]*Http
```

Reply:
if there is a persistent HTTP connection,  there could be multiple http flows flowing throw suricate, and suricate will send us the flows with same connection key but different TxID, given the TxID would be different for different flows but same TCP, we can use these TxIDs to check if different nodes are sending the same data and use these TxIDs to deduplicate these flows.